### PR TITLE
feature/JENKINS-46913-result-item-links * Linkify URLs in ResultItem headers

### PR DESCRIPTION
**Pull request by sophistifunk**
Linkify URLs in ResultItem headers

# Description

See [JENKINS-46913](https://issues.jenkins-ci.org/browse/JENKINS-46913).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by sophistifunk** [comment]
screen shot 2017-09-18 at 6 41 33 pm

File: jenkins-design-language/package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "url": "https://github.com/jenkinsci/jenkins-design-language.git"
   },
   "dependencies": {
+    "linkifyjs": "2.1.4",
```
Comment (michaelneale) [inline jenkins-design-language/package.json]: @sophistifunk this one? https://github.com/SoapBox/linkifyjs/blob/master/package.json so doesn't bring anything along, which is nice, right? (ie fairly small)

Comment (sophistifunk) [reply jenkins-design-language/package.json]: Yeah there's not a lot of code in there. It has a dependency on JQ, but only for functions we're not using so we can just ignore it.

**Comment by sophistifunk** [comment]
I can't seem to see what's actually going wrong here :-/

File: jenkins-design-language/src/js/components/ResultItem.jsx [hunk 2]
```diff
@@ -78,6 +79,11 @@ export class ResultItem extends Component {
         }
     };
 
+    urlClicked = (e: Event) => {
+        console.log('urlClicked');
```
Comment (cliffmeyers) [inline jenkins-design-language/src/js/components/ResultItem.jsx]: Probably just make sure we remove that before PR merge?

File: jenkins-design-language/package.json (hunk 1 above)
Comment (cliffmeyers) [reply jenkins-design-language/package.json]: I eyeballed the deps too and looks fine. I think as long as we verify that the bundles aren't bloating up after this change we are good. They do strange things sometimes :/

File: jenkins-design-language/src/js/components/ResultItem.jsx (hunk 2 above)
Comment (sophistifunk) [reply jenkins-design-language/src/js/components/ResultItem.jsx]: Nice catch, cheers

**Comment by michaelneale** [comment]
Nice - this works great. Wire it in and should go for it. Very nice.

**Comment by michaelneale** [comment]
@sophistifunk doesn't this need to be published/incorporated to take effect?

**Comment by sophistifunk** [comment]
@NicuPascu it's just to stop the expando from opening / closing when you click a link. It's the easiest way to do it.

**Comment by cliffmeyers** [comment]
Actually I think in the future we should be sure to publish every package change as part of the PR so it can through ATH. Otherwise we are just pushing that risk onto the next developer who actually needs to publish. Am quite confident this change won't be an issue, but.. famous last words :)

**Pull request merged**